highLevelInterface/pwaNloptFit.cc

```diff
@@ -25,13 +25,42 @@ double
 rpwaNloptFunc(unsigned int n, const double* x, double* gradient, void* func_data)
 {
 	const pwaLikelihood<complex<double> >* L = (const pwaLikelihood<complex<double> >*)func_data;
-	if(n != L->nmbPars()) {
+	if(n != L->nmbPars() - L->nmbParsFixed()) {
 		printErr << "parameter mismatch between NLopt and pwaLikelihood. Aborting..." << endl;
+		printErr << "Likelihood needs "  << L->nmbPars() << " parameters including " << L->nmbParsFixed() << " fixed parameters. Got " << n << " parameters" << endl;
 		throw;
 	}
+	if (L->nmbParsFixed() != 0) {
+		static double* xIncludingFixed = new double[L->nmbPars()];
```

Comment:
I think the code would be easier to read with the following name changes
n -> nmbParsNlopt
x -> parsNlopt
xIncludingFixed -> parsIncludingFixed

Reply:
I agree!